plugins/overprovisioning/overprovisioning/__init__.py

```diff
@@ -42,9 +42,15 @@ def deploy(
     if "cpu" not in parameters:
         parameters["cpu"] = "1"
 
+    if "node_group" in parameters:
+        node_group = parameters["node_group"]
+        del parameters["node_group"]
+    else: 
```

Comment:
I don't think we should default to this. primary-compute is a naming convention we've been using, but there is not guarantee anyone else will

Reply:
If it isn't filled out, then should we throw an error (make it required)?